File: app/src/main/java/mod/agus/jcoderz/editor/view/palette/IconMaterialSwitch.java
package mod.agus.jcoderz.editor.view.palette;

import android.content.Context;
import android.view.ViewGroup;

import com.besome.sketch.beans.LayoutBean;
import com.besome.sketch.beans.ViewBean;
import com.besome.sketch.editor.view.palette.IconBase;
import pro.sketchware.R;

import mod.agus.jcoderz.beans.ViewBeans;

public class IconMaterialSwitch extends IconBase {

    public IconMaterialSwitch(Context context) {
        super(context);
        setWidgetImage(R.drawable.ic_mtrl_toggle);
        setWidgetName("MaterialSwitch");
    }

    @Override
    public ViewBean getBean() {
        ViewBean viewBean = new ViewBean();
        viewBean.type = ViewBeans.VIEW_TYPE_WIDGET_MATERIALSWITCH;
        LayoutBean layoutBean = viewBean.layout;
        layoutBean.paddingLeft = 8;
        layoutBean.paddingTop = 8;
        layoutBean.paddingRight = 8;
        layoutBean.paddingBottom = 8;
        layoutBean.width = ViewGroup.LayoutParams.WRAP_CONTENT;
        viewBean.convert = getName();
        return viewBean;
    }
}

File: app/src/main/java/dev/juez/editor/view/palette/IconMaterialSwitch.kt
package dev.juez.editor.view.palette;

import android.content.Context;
import com.besome.sketch.beans.LayoutBean
import com.besome.sketch.beans.ViewBean
import com.besome.sketch.editor.view.palette.IconBase
import dev.juez.beans.JuezViewBeans
import pro.sketchware.R

class IconMaterialSwitch(private val context: Context): IconBase(context) {

  override fun a(context: Context) {
    super.a(context)
    setWidgetImage(R.drawable.ic_mtrl_toggle)
    setWidgetName("MaterialSwitch")
  }
  
  override fun getBean(): ViewBean {
    return ViewBean().apply {
      type = JuezViewBeans.VIEW_TYPE_WIDGET_MATERIALSWITCH
      text.text = getName()
      convert = "com.google.android.material.materialswitch.MaterialSwitch"
      layout.apply {
        paddingLeft = 8
        paddingTop = 8
        paddingRight = 8
        paddingBottom = 8
      }
    }
  }
}